docs/content/sips/006-spin-plugins.md

```diff
@@ -0,0 +1,233 @@
+### Usage
+
+The `spin plugin` command will have three sub-commands.
+
+```bash
+Commands for working with Spin plugins
+
+USAGE:
+    spin plugin <SUBCOMMAND>
+
+SUBCOMMANDS:
+    install      Install plugin as described by a remote or local plugin manifest
+    uninstall    Uninstall a plugin
+	update		 Update one or all plugins to the latest or specified version 
+```
+
+**`spin plugin install`**
+
+The `spin plugin install` subcommand installs a plugin named `$name`. By default, it will look for a plugin manifest named `$name.json` in the `spin-plugin` repository; however, it can be directed to use a local manifest or one at a different remote location using the `--file` or `--url` flag, respectively.
+
+```bash
+Install a Spin plugin using a plugin manifest file. 
+By default, looks for the plugin manifest named <name>.json
+in the Spin plugins repository https://github.com/fermyon/spin-plugins
+
+USAGE:
+    spin plugin install <name>
+
+OPTIONS:
+    -f, --file                       Path to local plugin manifest
+	-u, --url                        Address of remote plugin manifest
+    -v, --version                    Desired version to update plugin to. Defaults to latest.
+    -s, --skip-version-compat-check  Instructs Spin to install the plugin even 
+                                     if its manifest specifies that it is 
+                                     incompatible with the current version of Spin
+```
+
+If the manifest is found, Spin will check that the plugin is compatible with the current OS, platform, and version of Spin. If so, before installing the plugin, Spin will prompt the user as to whether to trust the source. For example, the following prompt would be displayed for a plugin named `deploy` with an Apache 2 license and hosted at [`https://github.com/fermyon/spin-plugin-deploy/releases/download/v0.1.0/spin-plugin-deploy-v0.1.0-macos-aarch64.tar.gz`](https://github.com/fermyon/spin-plugin-deploy/releases/download/v0.1.0/spin-plugin-deploy-v0.1.0-macos-aarch64.tar.gz):
+
+```bash
+Installing plugin deploy with license Apache 2.0 from https://github.com/fermyon/spin-plugin-deploy/releases/download/v0.1.0/spin-plugin-deploy-v0.1.0-macos-aarch64.tar.gz
+For more information, reference the plugin metadata at `https://github.com/fermyon/spin-plugins/plugin-manifests/deploy.json`.
+Are you sure you want to proceed? ('yes'/'no') (default: no)
+```
+
+The plugin will only be installed if a user enters `yes`. Otherwise, the command exits. 
+
+Spin will reference the plugin manifest in order to fetch the plugin binary and install it into the user’s local data directory under a Spin-managed `plugins` subdirectory. The plugin manifest will be stored within a `manifests` subdirectory. 
+
+After installing a plugin, it can be executed directly from the Spin CLI. For example, a plugin named `$name` would be executed by running `spin $name <args>`. Any additional arguments supplied will be passed when executing the associated binary.
```

Comment:
Can plugins override existing commands? For example, could I potentially write a custom `spin up` plugin that replaces the built-in version of `spin up`?

Reply:
I would say no. We can add a GitHub action to the plugins repository that prevents a naming conflict of that sort for plugins that live there and then have a check during install. Do you think a note on checking and denying installs of a plugin with the same name as an internal plugin should be added to the SIP?

Reply:
I think even if installed it cannot be invoked as the priority of the subcommands will be higher for the internal subcommands over the external ones but in any case it might be good to make sure such installations cannot take place.

Reply:
Okay! Can we add a comment in here stating that plugins matching an existing command name will not be accepted into spin-plugins? Or did we want to go even further and add a check to `spin plugin install` itself?

Reply:
I would think add a check to `spin plugin install` since plugins can exist outside of spin. I can add a note in the SIP in the install section.

Reply:
Note added in dbea6bf9a6b. I also changed the install example from `deploy` to `test` to not make the example contradictory to the note.